api/src/main/java/org/apache/iceberg/AddPartitionSpec.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg;
+
+import org.apache.iceberg.exceptions.CommitFailedException;
+import org.apache.iceberg.expressions.Expressions;
+import org.apache.iceberg.expressions.Term;
+
+/**
+ * API for partition spec evolution.
+ *
+ * <p>When committing, these changes will be applied to the current table metadata. Commit conflicts
+ * will not be resolved and will result in a {@link CommitFailedException}.
+ */
+public interface AddPartitionSpec extends PendingUpdate<PartitionSpec> {
```

Comment:
Instead of adding a fully new API couldn't we just add an API to UpdatePartitionSpec that just sets the starting point of the spec as unpartitioned?

Something like

```
updatePartitionSpec()
 .fromUnpartitioned()
 .add...
 .add..
 ```
 ?

Reply:
Yeah I think I can do that. Maybe from(spec)?

Reply:
I reverted the old commit and added a new one with an implementation of `fromSpec(..)`, and also added a test. PLMK if there's anything else I need to do for this to get merged 😄

Reply:
Tests are still failing at the moment, please check out the failed runs below

Reply:
Ok I think it should pass now, I've an ignore to the revapi, I hope that what I should have done 😄